lib/perl/Genome/Site/TGI/Extension/Sys.pm

```diff
@@ -630,7 +630,23 @@ sub get_mem_limit_from_bjobs {
     if ($bjobs_cmd && $LSB_JOBID) {
         chomp $bjobs_cmd;
         my $bjobs = qx($bjobs_cmd -l $LSB_JOBID);
-        my ($bjobs_mem_limit_kb) = $bjobs =~ /MEMLIMIT\s+(\d+)/;
+        my $bjobs_mem_limit_kb;
+        my ($bjobs_mem_limit, $bjobs_mem_limit_units) = $bjobs =~ /MEMLIMIT\s+(\d+\.?\d*)\s+([TGMK])\b/;
+        if ($bjobs_mem_limit_units eq 'T') {
+            $bjobs_mem_limit_kb = int($bjobs_mem_limit * 1024 * 1024 * 1024);
+        }
+        elsif ($bjobs_mem_limit_units eq 'G') {
+            $bjobs_mem_limit_kb = int($bjobs_mem_limit * 1024 * 1024);
+        }
+        elsif ($bjobs_mem_limit_units eq 'M') {
+            $bjobs_mem_limit_kb = int($bjobs_mem_limit * 1024);
+        }
+        elsif ($bjobs_mem_limit_units eq 'K') {
+            $bjobs_mem_limit_kb = int($bjobs_mem_limit);
+        }
+        else {
+            # failed to parse
+        }
         $mem_limit = $bjobs_mem_limit_kb if ($bjobs_mem_limit_kb);
     }
     return $mem_limit;
```

Comment:
The sub that calls it, mem_limit_kb, and the unit tests for that sub, allow for the possibility of undef being returned.  It seems there is the possibility that the job has no mem limit, and mem_limit_kb falls back to reporting on physical memory.

As for the extra variable...  Absolutely right.